packages/blockchain-wallet-v4-frontend/src/modals/Interest/InterestDetails/index.tsx

```diff
@@ -0,0 +1,82 @@
+import { actions } from 'data'
+import { bindActionCreators, compose, Dispatch } from 'redux'
+import { CoinType, RemoteDataType, SupportedCoinsType } from 'core/types'
+import { connect } from 'react-redux'
+import { getData } from './selectors'
+import { RatesType } from 'data/types'
+import DataError from 'components/DataError'
+import Loading from './template.loading'
+import React, { PureComponent } from 'react'
+import Success from './template.success'
+
+export type OwnProps = {
+  handleClose: () => void
+}
+
+export type LinkDispatchPropsType = {
+  interestActions: typeof actions.components.interest
+  simpleBuyActions: typeof actions.components.simpleBuy
+}
+
+export type SuccessStateType = {
+  coin: CoinType
+  rates: RatesType
+  supportedCoins: SupportedCoinsType
+}
+
+type LinkStatePropsType = {
+  data: RemoteDataType<string | Error, SuccessStateType>
+}
+
+type Props = OwnProps & LinkDispatchPropsType & LinkStatePropsType
+
+class InterestForm extends PureComponent<Props> {
+  state = {}
+
+  handleSimpleBuyClick = () => {
+    this.props.simpleBuyActions.showModal('sideNav')
+  }
+
+  handleDepositClick = () => {
+    this.props.interestActions.showInterestModal('DEPOSIT')
+  }
+
+  handleRefresh = () => {
+    this.props.interestActions.showInterestModal('DEPOSIT')
+  }
+
+  render () {
+    const { data } = this.props
+    return data.cata({
+      Success: val => (
+        <Success
+          {...val}
+          {...this.props}
+          handleDepositClick={this.handleDepositClick}
+          handleSimpleBuyClick={this.handleSimpleBuyClick}
+        />
+      ),
+      Failure: () => <DataError onClick={this.handleRefresh} />,
```

Comment:
If the details page fails to load refresh should take users to the DEPOSIT step? Or should we try to refetch w/e info failed to load?